src/main.rs

```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
     }
 }
 
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
```

Comment:
The expected behaviour has been clarified [here](https://github.com/gyng/rcanary/pull/4#discussion_r83370871).